gitHub.js

```diff
@@ -1,37 +1,100 @@
+const readline = require('readline');
 const dayjs = require("dayjs");
 require('dotenv').config();
 
 const today = dayjs().format("YYYY-MM-DD");
 const weekAgo = dayjs().subtract(8, 'days').format("YYYY-MM-DD");
+const monthAgo = dayjs().subtract(1, 'month').format("YYYY-MM-DD");
 
-async function getCommitsSince(user, timeFrame) {
-    let url;
-    if (timeFrame === "one-week-ago") {
-        url = `https://api.github.com/search/commits?q=author:${user}+committer-date:${weekAgo}..${today}`;
+async function getCommitsSince(user, currentUser, totalUsers) {
+    consoleOutput(`Processing ${currentUser + 1}/${totalUsers} users...`);
+
+    if (!await verifyUser(user)) return ['User does not exist.'];
+
+    return [await getWeekAgo(user), await getFourWeekAvg(user)];
+}
+
+async function verifyUser(user) {
+    if (!user) return false;
+
+    const res = await fetch(`https://api.github.com/search/users?q=${user}`, {
```

Comment:
I tried using `/users` like you suggested but that turned out to have a rate limit of 60 per hour. So I thought we can't be waiting for an hour between rate limit resets. What do you suggest?

Reply:
A few things that occur to me:
- Does `/search/commits` require an exact match for the `author`? If so, if it always gives `"total_results": 0`when the author doesn't exist, maybe _then_ we check if that user exists. Assuming most already exist and have >0 commits, that should keep the number of requests sensible.
- Could we check in the search results for an exact name match, rather than assuming `> 0` results means it must be in there?

Reply:
- The endpoint gives `> 0` results for some non-existing users so I don't think this way is practical.
- For the second suggestion, I see the proper username in `data.items[0].author.login` and it's always the correct username for the existing trainees and for the non-existing it's always wrong so I think it can be done this way.

**Non existing user:**
![image](https://github.com/CodeYourFuture/github-tracker/assets/87202358/ee5eadf2-ccdf-406a-aa56-f4249cfb9a82)

**Existing user:**
![image](https://github.com/CodeYourFuture/github-tracker/assets/87202358/1a8b3751-5051-4769-90f0-d4fca11a488d)

Reply:
I would [`.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) rather than only checking the first, that seems like a brittle assumption. But otherwise yes.